pilot/pkg/bootstrap/server.go

```diff
@@ -1072,7 +1078,9 @@ func (s *Server) secureGrpcStart(listener net.Listener) {
 		caCertPool.AppendCertsFromPEM(caCert)
 
 		opts = append(opts, grpc.Creds(creds))
+		s.secureGrpcMutex.Lock()
 		s.secureGRPCServer = grpc.NewServer(opts...)
```

Comment:
How can this happen?

Reply:
Removing this protection indeed doesn't make any difference. Not sure why I added it in the first place. I will revert.